feat: connext

erkarl opened the pull request:
Adds connext and

Closes https://github.com/ExchangeUnion/xud/issues/1457
Closes https://github.com/ExchangeUnion/xud-simnet/issues/78
Closes https://github.com/ExchangeUnion/xud-docker/issues/382
Closes https://github.com/ExchangeUnion/xud-docker/issues/277
Closes https://github.com/ExchangeUnion/xud-docker/issues/409
Closes https://github.com/ExchangeUnion/xud-docker/issues/425

Commit:
Add Connext support

reliveyy committed:
merge connext config

Commit:
Use connext branch for xud

kilrau reviewed (dismissed):
DELETE

kilrau commented:
`getbalance` doesn't show connext:
```
simnet > getbalance

Balance:
┌──────────┬─────────────────┬──────────────────────────────────┬───────────────────────────────┐
│ Currency │ Total Balance   │ Channel Balance (Tradable)       │ Wallet Balance (Not Tradable) │
├──────────┼─────────────────┼──────────────────────────────────┼───────────────────────────────┤
│ BTC      │ 146230.49681266 │ 1.4996285 (13.99756601 inactive) │ 146214.99961815               │
├──────────┼─────────────────┼──────────────────────────────────┼───────────────────────────────┤
│ LTC      │ 365291.98816702 │ 3.00891402 (28.979864 inactive)  │ 365259.999389                 │
└──────────┴─────────────────┴──────────────────────────────────┴───────────────────────────────┘
```

kilrau commented:
Steps to test in existing environment:
- get funds
- `addcurrency ETH Connext 18 --token_address="0x0000000000000000000000000000000000000000"`
- `addpair eth/btc`
- `buy 0.001 ETH/BTC 1`

kilrau commented:
Current result as per testing with @erkarl :+1: swap failing due to a time out:
```
22/04/2020 15:34:31.606 [P2P] debug: received swapRequest from 0231d0a3e27c4db93cf070255d834c4ec1bf7eafd1bf6652b79724c47623ad0acf (CleanTraffic): {"proposedQuantity":100000,"pairId":"ETH/BTC","orderId":"9ff8f390-84ae-11ea-b6b3-6fbbe5142802","rHash":"69bffbb767fa8904e7338977412b9cf4d7a8278f8091d2b73fa6b56b2ef49392","takerCltvDelta":40}
22/04/2020 15:34:31.607 [ORDERBOOK] debug: added hold of 100000 on order 9ff8f390-84ae-11ea-b6b3-6fbbe5142802
22/04/2020 15:34:31.622 [SWAPS] debug: trying to accept deal: {"quantity":100000,"localId":"9ff8f390-84ae-11ea-b6b3-6fbbe5142802","price":1,"isBuy":true} from xudPubKey: 0231d0a3e27c4db93cf070255d834c4ec1bf7eafd1bf6652b79724c47623ad0acf
22/04/2020 15:34:31.623 [SWAPS] debug: New deal: {"proposedQuantity":100000,"pairId":"ETH/BTC","orderId":"9ff8f390-84ae-11ea-b6b3-6fbbe5142802","rHash":"69bffbb767fa8904e7338977412b9cf4d7a8278f8091d2b73fa6b56b2ef49392","takerCltvDelta":40,"price":1,"isBuy":true,"quantity":100000,"makerAmount":100000,"takerAmount":100000,"makerCurrency":"ETH","takerCurrency":"BTC","makerUnits":1000000000000000,"takerUnits":100000,"takerPubKey":"038cc25e0338c7d75f7745ee8918e6d781b97cfc62a1017fa3b4314e30ac83c04f","destination":"038cc25e0338c7d75f7745ee8918e6d781b97cfc62a1017fa3b4314e30ac83c04f","peerPubKey":"0231d0a3e27c4db93cf070255d834c4ec1bf7eafd1bf6652b79724c47623ad0acf","localId":"9ff8f390-84ae-11ea-b6b3-6fbbe5142802","phase":0,"state":0,"role":1,"createTime":1587569671623}
22/04/2020 15:34:31.631 [LND-BTC] debug: found a route to 038cc25e0338c7d75f7745ee8918e6d781b97cfc62a1017fa3b4314e30ac83c04f for 100000 units with finalCltvDelta 40: 19833,,100000,21746140974219264,100000000,100000,,19833,100000000,,038cc25e0338c7d75f7745ee8918e6d781b97cfc62a1017fa3b4314e30ac83c04f,,100000000
22/04/2020 15:34:31.671 [SWAPS] debug: got BTC block height of 19793
22/04/2020 15:34:31.672 [SWAPS] debug: found route to taker with total lock duration of 40 BTC blocks (~7h)
22/04/2020 15:34:31.674 [SWAPS] debug: calculated lock buffer for first leg: 4854 ETH blocks (~20h)
22/04/2020 15:34:31.674 [SWAPS] debug: lock delta for final hop to maker: 5054 ETH blocks (~21h)
22/04/2020 15:34:31.707 [SWAPS] debug: sending swap accepted packet: {"makerCltvDelta":5054,"rHash":"69bffbb767fa8904e7338977412b9cf4d7a8278f8091d2b73fa6b56b2ef49392","quantity":100000} to peer: 0231d0a3e27c4db93cf070255d834c4ec1bf7eafd1bf6652b79724c47623ad0acf
22/04/2020 15:34:36.095 [RPC] debug: received call /xudrpc.Xud/ListPeers
22/04/2020 15:34:56.095 [RPC] debug: received call /xudrpc.Xud/ListPeers
22/04/2020 15:35:01.624 [SWAPS] debug: deal 69bffbb767fa8904e7338977412b9cf4d7a8278f8091d2b73fa6b56b2ef49392 failed due to SwapTimedOut
22/04/2020 15:35:01.627 [ORDERBOOK] debug: removed hold of 100000 on order 9ff8f390-84ae-11ea-b6b3-6fbbe5142802
22/04/2020 15:35:01.628 [SWAPS] debug: Sending swap error to peer: {"rHash":"69bffbb767fa8904e7338977412b9cf4d7a8278f8091d2b73fa6b56b2ef49392","failureReason":10}
22/04/2020 15:35:01.923 [P2P] verbose: received order from 0231d0a3e27c4db93cf070255d834c4ec1bf7eafd1bf6652b79724c47623ad0acf (CleanTraffic): {"id":"c2bfd830-84ae-11ea-9e1e-b17f8a74c49a","pairId":"ETH/BTC","price":1,"quantity":100000,"isBuy":false}
22/04/2020 15:35:01.924 [ORDERBOOK] debug: order added: {"id":"c2bfd830-84ae-11ea-9e1e-b17f8a74c49a","pairId":"ETH/BTC","price":1,"quantity":100000,"isBuy":false,"peerPubKey":"0231d0a3e27c4db93cf070255d834c4ec1bf7eafd1bf6652b79724c47623ad0acf","createdAt":1587569701924,"initialQuantity":100000}
```

Commit:
dynamic webhookport

sangaman reviewed (commented):
Just gave this a spin, the swap is failing but at least I was able to get to that point.

My steps using an existing simnet node in docker:

1. connect to simnet node if not already connected `connect 03193508d52dc8235c1bf6e902dae30c6dc1823f1e3cdc3c1a97652af9142cd4fe@xud1.simnet.exchangeunion.com:28885`
1. `addcurrency ETH Connext 18 --token_address="0x0000000000000000000000000000000000000000"`
1. `addpair ETH BTC`
1. `restart xud`
1. `getinfo` - copy the connext address (as an aside, I'm thinking we should return this address when we call `deposit ETH`
1. get eth from faucet: `curl -H "Content-Type: application/json" -X POST -d '{"address":"0xFDe0b13b271C335cac257F3536FEeEc491286A01"}' http://35.193.71.174:9000/faucet`
1. `openchannel Connext ETH 2`
1. `orderbook` - display available orders
1. `sell 0.0001 ETH/BTC mkt` to match an existing buy order

Swap failed, log output below:

```
22/04/2020 19:13:49.017 [SWAPS] debug: Requesting deal: {"takerCltvDelta":40,"rHash":"18ad58d57d7a738de66208e84cd5efec70199d7e0c638c94d88a6545b382e9ed","orderId":"9ff8f390-84ae-11ea-b6b3-6fbbe5142802","pairId":"ETH/BTC","proposedQuantity":10000,"rPreimage":"a8b61e32679a865e305afcd84031e1eb2e1e42d530f7a80b7dacec5b6307a60c","takerCurrency":"BTC","makerCurrency":"ETH","takerAmount":10000,"makerAmount":10000,"takerUnits":10000,"makerUnits":100000000000000,"destination":"indra7FTKMbFauqFeAYkEeQ4KiEcfGxUNXMbdCEPRGaMvRBEYwSHGRx","peerPubKey":"03193508d52dc8235c1bf6e902dae30c6dc1823f1e3cdc3c1a97652af9142cd4fe","localId":"6523d270-84cd-11ea-923e-27c70e631bb7","price":1,"isBuy":true,"phase":0,"state":0,"role":0,"createTime":1587582829010}
22/04/2020 19:13:49.925 [P2P] debug: received swapAccepted from 03193508d52dc8235c1bf6e902dae30c6dc1823f1e3cdc3c1a97652af9142cd4fe (MeltWedding): {"rHash":"18ad58d57d7a738de66208e84cd5efec70199d7e0c638c94d88a6545b382e9ed","quantity":10000,"makerCltvDelta":5054}
22/04/2020 19:13:49.981 [LND-BTC] debug: added invoice of 10000 for 18ad58d57d7a738de66208e84cd5efec70199d7e0c638c94d88a6545b382e9ed with cltvExpiry 40
22/04/2020 19:14:19.563 [P2P] debug: received swapFailed from 03193508d52dc8235c1bf6e902dae30c6dc1823f1e3cdc3c1a97652af9142cd4fe (MeltWedding): {"rHash":"18ad58d57d7a738de66208e84cd5efec70199d7e0c638c94d88a6545b382e9ed","failureReason":10}
22/04/2020 19:14:19.564 [SWAPS] debug: deal 18ad58d57d7a738de66208e84cd5efec70199d7e0c638c94d88a6545b382e9ed failed due to SwapTimedOut
```

erkarl commented:
@sangaman can you ensure your `~/.xud-docker/simnet/data/xud/xud.conf` for http service is the following:
```
[connext]
disable = false
host = "connext"
port = 5040
webhookhost = "xud"
webhookport = 28887
```
I updated the image, but if you had an existing data directory it wouldn't overwrite the configuration. Previously, the webhookport value was 8887.

kilrau commented:
Also to be fixed:
```
simnet > getinfo

General XUD Info
┌───────────────┬───────────────────────────────────────────────────────────────────────┐
│ Status        │ Raiden: Could not fetch status informatio                             │
├───────────────┼───────────────────────────────────────────────────────────────────────┤
```

reliveyy committed:
test: shorten simnet channel waiting timeout

kilrau commented:
Fixes to be done @reliveyy 
- xud should show `Ready` according to the same logic as on testnet/mainnet
- connext should show ready if connext client is up and responding to a call. @erkarl which call should we use?

```
simnet > status
┌─────────┬──────────────────────────────────────────────────┐
│ SERVICE │ STATUS                                           │
├─────────┼──────────────────────────────────────────────────┤
│ lndbtc  │ Ready                                            │
├─────────┼──────────────────────────────────────────────────┤
│ lndltc  │ Ready                                            │
├─────────┼──────────────────────────────────────────────────┤
│ connext │ Container running                                │
├─────────┼──────────────────────────────────────────────────┤
│ xud     │ Container running                                │
└─────────┴──────────────────────────────────────────────────┘
```

- [ ] https://github.com/ExchangeUnion/xud-docker/issues/430 which is VERY annoying

erkarl commented:
> connext should show ready if connext client is up and responding to a call. @erkarl which call should we use?

You can use `http://connextHost:connextPort/health` that will respond with 204 status code when it's up and running.

reliveyy committed:
fix long-time stopping issue (resolves #430)

reliveyy committed:
fix connext status

reliveyy committed:
test: extend channel waiting timeout to 1000 seconds

reliveyy committed:
test: remove raiden assertions

kilrau commented:
More fixes for @reliveyy : 

- [ ] when calling status immediately after `xud ctl` was opened its layout and connext client status is broken:
```
simnet > status
┌─────────┬──────────────────────────────────────────────────┐
│ SERVICE │ STATUS                                           │
├─────────┼──────────────────────────────────────────────────┤
│ lndbtc  │ fetching..                                       │
├─────────┼─Container running                               ─┤
│ lndltc  │ Container running                                │
├─────────┼──────────────────────────────────────────────────┤
│ connext │ 7|''                                             │
├─────────┼──────────────────────────────────────────────────┤
│ xud     │ xud is starting... try again in a few seconds    │
  ────────┴──────────────────────────────────────────────────┘
```
(some seconds later it's fine)
```
simnet > status
┌─────────┬──────────────────────────────────────────────────┐
│ SERVICE │ STATUS                                           │
├─────────┼──────────────────────────────────────────────────┤
│ lndbtc  │ Ready                                            │
├─────────┼──────────────────────────────────────────────────┤
│ lndltc  │ Ready                                            │
├─────────┼──────────────────────────────────────────────────┤
│ connext │ Ready                                            │
├─────────┼──────────────────────────────────────────────────┤
│ xud     │ Waiting for channels                             │
└─────────┴──────────────────────────────────────────────────┘
```

kilrau commented:
Other than that getting channels for all 3 currencies now :tada: 

```
simnet > getbalance

Balance:
┌──────────┬───────────────┬────────────────────────────┬───────────────────────────────┐
│ Currency │ Total Balance │ Channel Balance (Tradable) │ Wallet Balance (Not Tradable) │
├──────────┼───────────────┼────────────────────────────┼───────────────────────────────┤
│ BTC      │ 0.5           │ 0 (0.5 pending)            │ 0                             │
├──────────┼───────────────┼────────────────────────────┼───────────────────────────────┤
│ ETH      │ 2.49979       │ 1                          │ 1.49979                       │
├──────────┼───────────────┼────────────────────────────┼───────────────────────────────┤
│ LTC      │ 1             │ 0 (1 pending)              │ 0                             │
└──────────┴───────────────┴────────────────────────────┴───────────────────────────────┘
```

reliveyy committed:
add defaultchanconfs

reliveyy committed:
update lnd uris

Commit:
use master branch of xud for xud:latest

reliveyy committed:
fix 7|''

reliveyy committed:
adjust connext status error text

michael1011 committed:
fix: xud simnet reconnection issue

reliveyy committed:
avoid lnd connecting to itself

kilrau committed:
fix typo: staring -> starting

reliveyy committed:
skip simnet test

kilrau commented:
Here again for clarity @reliveyy :

- [ ] status format still breaking seconds after `up`:
```
simnet > status
┌─────────┬──────────────────────────────────────────────────┐
│ SERVICE │ STATUS                                           │
├─────────┼──────────────────────────────────────────────────┤
│ lndbtc  │ Container running                                │
├─────────┼──────────────────────────────────────────────────┤
│ lndltc  │ fetching..                                       │
├─────────┼─Container running                               ─┤
│ connext │ fetching..                                       │
├─────────┼─connext is starting... try again in a few second─┤
│ xud     │ xud is starting... try again in a few seconds    │
  ────────┴──────────────────────────────────────────────────┘
```

- [ ] enhance https://github.com/ExchangeUnion/xud-docker/pull/429/files#diff-b5bc39c8275a1de7af941b22d014795a to **break the loop** (trying to connect) if call returns `already connected to peer`

kilrau committed:
fix: lndbtc port typo 29375 -> 29735, lndltc port typo 30375 -> 30735

kilrau committed:
fix: more port fixes

reliveyy commented on images/lnd/0.9.0-beta-ltc-patched/entrypoint.sh:
```diff
@@ -37,15 +37,20 @@ echo "[entrypoint] Onion address for lndbtc is $LND_ADDRESS"
-    local key="023f670b916d8b89e362f4832f8eeca4f2d578a737c97f6fd4845bb7b584647667"
-    local uri="$key@xud1.simnet.exchangeunion.com:10011"
+    local key="03af9876d7a367cd92d192c745194e91173830c9e4f0d51cfe19beb687189a99e5"
+    local uri="$key@zqnr7vepxwk6w26zlmuh55edg45yu44rcqdk5d2gzfp7rtodduc6ycid.onion:30375"
```
The URI port is still 375 (not 735)？

reliveyy commented on images/lnd/0.9.2-beta-patched/entrypoint.sh:
```diff
@@ -36,15 +36,20 @@ LND_ADDRESS=$(cat "$LND_HOSTNAME")
-    local key="02db09dd366d7ba6d061502b5b6db1bbb47c0daacd36fc399ab617fd6406cf822a"
-    local uri="$key@xud1.simnet.exchangeunion.com:10012"
+    local key="0331c98ab6ec199164130cca2b2118a17e7d5e8b53d4e46ddfd4b656a30c636b70"
+    local uri="$key@5ucnksx35jjj34atausbifhyoiv7adsrjatqb5vpns6fftqaxmbvzfid.onion:29375"
```
Same here 375?

reliveyy committed:
shorten connext and xud status to "starting..."

reliveyy committed:
add newline

reliveyy committed:
break loop when "already connected to peer"

kilrau committed:
fix: xud1 tor -> clearnet

kilrau commented:
For documentation purposes: I made local1_lndbtc, xud1_lndbtc, local2_lndbtc all connect to our gcloud btcd instead of using neutrino, lndbtc's have tor natively enabled, just like before, synced local1/2 from scratch. Result: trying to pay immediately after synced to chain/graph are true results in the same failures (someimtes for both sides, but mostly only for one). After some 10 more minutes it magically starts working.

```
simnet > lndbtc-lncli payinvoice lnsb10n1p02stwypp5cyf2gt3tlpk6twq050tyn6s3udln9mvkzx88qsr8q0rzpdun07tqdqqcqzpgsp5n5hfe37gajrhvdnjgkvaccx8wp950lvjuq469tgccnc49s4z4dkq9qy9qsq0pjthx6glsfm6025twa7nz4k9em38l7hcss3pthmjvzquuavlweh75dqp8g6wn3cm0nrmh6g9ejnkdjdtjk9xr0sj8da9l5pd2dg8gqpl40krt
Description: 
Amount (in satoshis): 1
Destination: 0392f058255e7efa0b60fe4db7b9f8062cd37b89692170499556244609b557dd69
Confirm payment (yes/no): yes
{
    "payment_error": "insufficient local balance",
    "payment_preimage": null,
    "payment_route": null,
    "payment_hash": "c112a42e2bf86da5b80fa3d649ea11e37f32ed96118e70406703c620b7937f96"
}
[lncli] insufficient local balance
simnet > lndbtc-lncli payinvoice lnsb10n1p02stwypp5cyf2gt3tlpk6twq050tyn6s3udln9mvkzx88qsr8q0rzpdun07tqdqqcqzpgsp5n5hfe37gajrhvdnjgkvaccx8wp950lvjuq469tgccnc49s4z4dkq9qy9qsq0pjthx6glsfm6025twa7nz4k9em38l7hcss3pthmjvzquuavlweh75dqp8g6wn3cm0nrmh6g9ejnkdjdtjk9xr0sj8da9l5pd2dg8gqpl40krt
Description: 
Amount (in satoshis): 1
Destination: 0392f058255e7efa0b60fe4db7b9f8062cd37b89692170499556244609b557dd69
Confirm payment (yes/no): yes
{
    "payment_error": "insufficient local balance",
    "payment_preimage": null,
    "payment_route": null,
    "payment_hash": "c112a42e2bf86da5b80fa3d649ea11e37f32ed96118e70406703c620b7937f96"
}
[lncli] insufficient local balance
simnet > lndbtc-lncli payinvoice lnsb10n1p02stwypp5cyf2gt3tlpk6twq050tyn6s3udln9mvkzx88qsr8q0rzpdun07tqdqqcqzpgsp5n5hfe37gajrhvdnjgkvaccx8wp950lvjuq469tgccnc49s4z4dkq9qy9qsq0pjthx6glsfm6025twa7nz4k9em38l7hcss3pthmjvzquuavlweh75dqp8g6wn3cm0nrmh6g9ejnkdjdtjk9xr0sj8da9l5pd2dg8gqpl40krt
Description: 
Amount (in satoshis): 1
Destination: 0392f058255e7efa0b60fe4db7b9f8062cd37b89692170499556244609b557dd69
Confirm payment (yes/no): yes
{
    "payment_error": "insufficient local balance",
    "payment_preimage": null,
    "payment_route": null,
    "payment_hash": "c112a42e2bf86da5b80fa3d649ea11e37f32ed96118e70406703c620b7937f96"
}
[lncli] insufficient local balance
simnet > lndbtc-lncli payinvoice lnsb10n1p02stwypp5cyf2gt3tlpk6twq050tyn6s3udln9mvkzx88qsr8q0rzpdun07tqdqqcqzpgsp5n5hfe37gajrhvdnjgkvaccx8wp950lvjuq469tgccnc49s4z4dkq9qy9qsq0pjthx6glsfm6025twa7nz4k9em38l7hcss3pthmjvzquuavlweh75dqp8g6wn3cm0nrmh6g9ejnkdjdtjk9xr0sj8da9l5pd2dg8gqpl40krt
Description: 
Amount (in satoshis): 1
Destination: 0392f058255e7efa0b60fe4db7b9f8062cd37b89692170499556244609b557dd69
Confirm payment (yes/no): yes
{
    "payment_error": "unable to find a path to destination",
    "payment_preimage": null,
    "payment_route": null,
    "payment_hash": "c112a42e2bf86da5b80fa3d649ea11e37f32ed96118e70406703c620b7937f96"
}
[lncli] unable to find a path to destination
simnet > lndbtc-lncli payinvoice lnsb10n1p02stwypp5cyf2gt3tlpk6twq050tyn6s3udln9mvkzx88qsr8q0rzpdun07tqdqqcqzpgsp5n5hfe37gajrhvdnjgkvaccx8wp950lvjuq469tgccnc49s4z4dkq9qy9qsq0pjthx6glsfm6025twa7nz4k9em38l7hcss3pthmjvzquuavlweh75dqp8g6wn3cm0nrmh6g9ejnkdjdtjk9xr0sj8da9l5pd2dg8gqpl40krt
Description: 
Amount (in satoshis): 1
Destination: 0392f058255e7efa0b60fe4db7b9f8062cd37b89692170499556244609b557dd69
Confirm payment (yes/no): yes
{
    "payment_error": "unable to find a path to destination",
    "payment_preimage": null,
    "payment_route": null,
    "payment_hash": "c112a42e2bf86da5b80fa3d649ea11e37f32ed96118e70406703c620b7937f96"
}
[lncli] unable to find a path to destination
```

raladev commented:
https://github.com/ExchangeUnion/xud-docker/pull/441#issuecomment-621149177
issue with status command and Ready status.

How it should be:
- % <100% --> Syncing 13.37% (5634/29633) 
- % =100% --> Ready

kilrau committed:
apply create/restore flow for simnet (#441)

* apply create/restore flow for simnet

* enable xud-backup for simnet

* remove --noseedbackup for lnds

* remove admin.macaroon waiting

* remove -m option to let background jobs respond to SIGINT

Setting -m will let background jobs run in another process group which will ignore SIGINT. Removing -m will let the jobs run in the same process group.
Ref. https://stackoverflow.com/questions/45106725/why-do-shells-ignore-sigint-and-sigquit-in-backgrounded-processes

* add 5 seconds waiting before create/restore flow to prevent lnd unlock stuck

* fix lndltc version missing

* fix typo

* extend lnd warn up waiting to 15 seconds

* beautify starting...

* restart lnds after successful xucli create

* add syncing details to lnd status

* reduce staring... waiting time to 5 seconds

* add 5 seconds waiting before restarting lnds

* change starting... to warming up...

* move restarts after backup location setup & add xud restart

* comment out xud restart

* waiting for tls.cert

* restart xud with 15 seconds delay

* remove ltcd and raiden dirs

* block restarting until xud is locked

* auto unlock

* refine lnd restarts

* refine unlock

* comment out lncli connect

* adjust restart text

* fix socket timeout when exec_run

* adjust restart text

* refine restarting text

* refine auto-unlock with Ctrl-C and pre-check for xud lock status

Co-authored-by: Yang Yang <reliveyy@gmail.com>

kilrau committed:
test: reduce trickledelay (graph update share interval) to 1 second

kilrau commented:
Can you change the connext image tag from `exchangeunion/connext:1.0.0-beta.2-simnet` -> `exchangeunion/connext:6.3.1` (whatever the  version is) with the next time you push a new image? @erkarl

kilrau committed:
chore: trigger build

kilrau committed:
fix: set lndbtc/lndltc chanconfs to 6 since it takes 6 confs to receive payments

kilrau committed:
Upgrade lndbtc to 0.10.0 (#444)

* try with lnd 0.10.0-beta

* revert lndltc to 0.9.0-beta

Co-authored-by: Yang Yang <reliveyy@gmail.com>

Commit:
temp: use fix/connext-v2 branch for xud:latest

Commit:
lock connext container down to specific hash

Commit:
change connext node and ethprovider ip

kilrau commented:
TODO:
- [ ] change connext image tag from exchangeunion/connext:1.0.0-beta.2-simnet -> exchangeunion/connext:6.3.1
- [ ] connext/connext swaps succeed, but have weird error output (please add @raladev )

raladev commented:
- [ ] connext/connext swaps succeed, but have weird error output
- perform XUC/ETH swap -> swap is successful, but error appears in xud logs):
`07/05/2020 12:05:31.935 [CONNEXT] error: ignoring received pending transfer because it does not meet the requirements -
        expectedUnits: 1000000000000000000 actualUnits: 1000000000000000000,
        expectedHash: 015bc932be8bbc1465d5addbce7959cd7ab41f4d5357c52f1b432b76e3305fd5 actualHash: 015bc932be8bbc1465d5addbce7959cd7ab41f4d5357c52f1b432b76e3305fd5,
        expectedTokenAddress: 0x0000000000000000000000000000000000000000 actualTokenAddress: 0xEcFcaB0A285d3380E488A39B4BB21e777f8A4EaC,
        expectedTimeLock: 200 actualTimelock: 640`
- [ ] auto-balancing lead to error 500 
after my swap autobalancing was triggered and I got a lot of errors in xud and connext and did not got new balance.
[connext_log.txt](https://github.com/ExchangeUnion/xud-docker/files/4594594/connext_log.txt)
[xud_log.txt](https://github.com/ExchangeUnion/xud-docker/files/4594595/xud_log.txt)
- [ ] Connext can not be disabled through xud config. 
I tried to disable connext through ~/.xud-docker/simnet/data/xud/xud.conf (stop xud container -> change param to disable - true -> start container again), but after running the value in config were changed to false without my interaction -_-
- [ ] Just a note: listcurrencies XUC contract address is 0xEcFcaB0A285d3380E488A39B4BB21e777f8A4EaC (For mainnet in should be 0xc324a2f6b05880503444451b8b27e6f9e63287cb)
- [ ] low prio: strange warn message for taker when swap
`07/05/2020 11:24:07.191 [CONNEXT] warn: received unexpected incoming transfer with rHash 178f3f3d299212cfd843be0cad1c8c086dd4e5e899bb83ae6539a7412a05e3ae
Warn during the trade for tater`
- Question: after backup recovering I got the same address and the same balance for ETH and XUC (but not for BTC/LTC, because I think I again got typo in passphrase and it is already starting to look suspicious and annoying for me). How the recovering works for Connext? only backup file of xud should be provided to get access to wallet?

erkarl commented:
> Question: after backup recovering I got the same address and the same balance for ETH and XUC (but not for BTC/LTC, because I think I again got typo in passphrase and it is already starting to look suspicious and annoying for me). How the recovering works for Connext? only backup file of xud should be provided to get access to wallet?

The seed for connext is derived from xud's seed + the `connext-store` folder in `connext` container should be backed up. This has not been implemented, yet.

erkarl commented:
> auto-balancing lead to error 500

Fixed by https://github.com/ExchangeUnion/xud/pull/1540

Commit:
Revert "temp: use fix/connext-v2 branch for xud:latest"

This reverts commit b29d140bf027bb8b9165cde05e278bea44f55c5a.

kilrau commented:
> The seed for connext is derived from xud's seed + the `connext-store` folder in `connext` container should be backed up. This has not been implemented, yet.
> 
> I'd handle the restore/backup separately from this PR.

Agree, not important for "connext on simnet", opened https://github.com/ExchangeUnion/xud/issues/1545 - let's handle this in the course of bringing connext to testnet

kilrau commented:
> Question: after backup recovering I got the same address and the same balance for ETH and XUC (but not for BTC/LTC, because I think I again got typo in passphrase and it is already starting to look suspicious and annoying for me). How the recovering works for Connext? only backup file of xud should be provided to get access to wallet?

Would you mind testing restore for lndbtc/lndltc again properly? @raladev This **has** to work flawlessly.

reliveyy committed:
add python3 to arm64 build for connext

reliveyy committed:
add make g++

reliveyy committed:
add python

reliveyy committed:
add python3 make g++ and python for amd64 connext builder stage

reliveyy committed:
tag connext as 6.3.8

raladev committed:
Change Container Running to Waiting for connext

because of #452

reliveyy committed:
change ConnextProject/ to connext/

Commit:
use connext:latest instead of 6.3.8

Commit:
temp: use fix/connext-minor-fixes branch for xud:latest

raladev commented:
- [] 500 error instead of ''400 insufficient funds to open channel" error for connext
- [] No node pub key in log (see second screen) 
[Screenshot from 2020-05-14 16-06-39](https://user-images.githubusercontent.com/29906866/81938043-10d6d280-95fd-11ea-95dd-779e02bde361.png)
[Screenshot from 2020-05-14 16-06-50](https://user-images.githubusercontent.com/29906866/81938038-0f0d0f00-95fd-11ea-9869-83b6ba82fd8b.png)

erkarl commented:
>     * [ ]  500 error instead of ''400 insufficient funds to open channel" error for connext

https://github.com/connext/rest-api-client/issues/23

>     * [ ]  No node pub key in log (see second screen)
>       [Screenshot from 2020-05-14 16-06-39](https://user-images.githubusercontent.com/29906866/81938043-10d6d280-95fd-11ea-95dd-779e02bde361.png)
>       [Screenshot from 2020-05-14 16-06-50](https://user-images.githubusercontent.com/29906866/81938038-0f0d0f00-95fd-11ea-9869-83b6ba82fd8b.png)

https://github.com/ExchangeUnion/xud/issues/1560

Commit:
Revert "temp: use fix/connext-minor-fixes branch for xud:latest"

This reverts commit cac067c42b77f677ce2fd6e2c234219b887b6d0e.
